File: 929 - Unique Email Addresses/Solution.java
import java.util.Set;
import java.util.HashSet;

class Solution {
    public int numUniqueEmails(String[] emails) {
        Set<String> set = new HashSet<>();
        for (String email: emails) {
            // Split the email into local and domain parts
            String[] parts = email.split("@");
            String local = parts[0];
            String domain = parts[1];

            // Remove '.' and anything after '+' in the local part
            local = local.replace(".", "");
            int plusIndex = local.indexOf("+");
            // if plus sign exists in the local parts
            if (plusIndex != -1)
            {
                local = local.substring(0, plusIndex);
            }

            // Add the cleaned email to the set
            set.add(local + "@" + domain);
        }
        return set.size();
    }
}

File: 929 - Unique Email Addresses/Solution.py
class Solution:
    def numUniqueEmails(self, emails: List[str]) -> int:
        # Create an empty set to store unique emails
        unique = set()

        # Iterate through the list of emails
        for email in emails:
            # Split the email into local and domain parts
            local, domain = email.split("@")
            # Remove anything after the '+' sign in the local part
            local = local.split("+")[0]
            # Replace '.' with empty string
            local = local.replace(".", "")
            # Add the cleaned email to the set
            unique.add((local, domain))
        # Return the length of the set, which is the number of unique emails
        return len(unique)